webapp/components/rhs_thread/rhs_thread.jsx

```diff
@@ -320,13 +322,23 @@ export default class RhsThread extends React.Component {
         return this.refs.postListContainer;
     }
 
+    handleClose() {
+        GlobalActions.emitCloseRightHandSide();
+        this.props.shrink();
+    }
+
     render() {
         if (this.props.posts == null || this.props.selected == null) {
             return (
                 <div/>
             );
         }
 
+        const isCurrentUserChannelMember = ChannelStore.isCurrentUserChannelMember(this.props.selected.channel_id);
+        if (!isCurrentUserChannelMember) {
+            this.handleClose();
```

Comment:
Noted. I added `GlobalActions.toggleSideBarAction(false);` to websocket action - `SocketEvents.USER_REMOVED`